src/components/app/AppLocalizationProvider.js

```diff
@@ -99,16 +99,69 @@ class AppLocalizationFetcher extends React.PureComponent<FetchProps> {
 
 type InitProps = {|
   +requestL10n: typeof requestL10n,
+  +requestedLocales: null | string[],
 |};
 
 /**
- * This component is responsible for initializing the locales.
+ * This component is responsible for initializing the locales as well as
+ * persisting the current locale to localStorage.
  */
 class AppLocalizationInit extends React.PureComponent<InitProps> {
   componentDidMount() {
     const { requestL10n } = this.props;
-    const locales = navigator.languages;
-    requestL10n(locales);
+    requestL10n(this._getPersistedLocale() ?? navigator.languages);
+  }
+
+  componentDidUpdate() {
+    this._persistCurrentLocale();
+  }
+
+  _getPersistedLocale(): string[] | null {
+    const strPreviouslyRequestedLocales =
+      localStorage.getItem('requestedLocales');
+
+    if (strPreviouslyRequestedLocales) {
+      try {
+        const previouslyRequestedLocales = JSON.parse(
+          strPreviouslyRequestedLocales
+        );
+        if (
+          Array.isArray(previouslyRequestedLocales) &&
+          previouslyRequestedLocales.length
+        ) {
+          return previouslyRequestedLocales;
+        }
+
+        console.warn(
+          `The stored locale information (${strPreviouslyRequestedLocales}) looks incorrect.`
+        );
+      } catch (e) {
+        console.warn(
+          `We got an error when trying to parse the previously stored locale information (${strPreviouslyRequestedLocales}).`
+        );
+      }
+    }
+
+    return null;
+  }
+
+  _persistCurrentLocale() {
+    const { requestedLocales } = this.props;
+    if (!requestedLocales) {
+      return;
+    }
+
+    try {
+      localStorage.setItem(
+        'requestedLocales',
+        JSON.stringify(requestedLocales)
+      );
+    } catch (e) {
+      console.warn(
+        'We got an error when trying to save the current requested locale. We may run in private mode.',
```

Comment:
Is this still correct for locale storage? I can currently change it and it persists as long as you don't close the window.

Reply:
That's true in some browsers but not Firefox :-)